docs: python openapi client (#1711)
treeverse/lakeFS: docs/using/python.md

# Calling the lakeFS API from Python
{: .no_toc }

The [lakeFS API](../reference/api.md){: target="_blank" } is OpenAPI 3.0 compliant, allowing the generation of clients from multiple languages.

> For Python, this example uses [OpenAPI Generator](hhttps://openapi-generator.tech){: target="_blank" }

Reviewer: ```suggestion
For Python, this example uses [OpenAPI Generator](https://openapi-generator.tech){: target="_blank" }
```

> which can generates Python client code for the OpenAPI definition served by a lakeFS server.

Reviewer: ```suggestion
that generates Python client code for the OpenAPI definition served by a lakeFS server.
```

> The lakefs-client pacakge was created by [OpenAPI Generator](https://openapi-generator.tech){: target="_blank" } using our OpenAPI definition served by a lakeFS server.

Reviewer: I would add a note that this is the recommended client package for python, but other solutions are valid too (Bravado, simple HTTP client)

Author: I'll add a note that lakeFS can be accessed using any OpenAPI v3 library or directly by HTTP client.
Bravado can't be used - supports only OpenAPI v1 and v2.

## Table of contents
{: .no_toc .text-delta }

1. TOC
{:toc}

## Install lakeFS Python Client API

Install the Python client using pip:

```shell
pip install 'lakefs==<lakeFS version>'
```

> The package is available from version 0.34.0.

Reviewer: Where did you come up with this version?

Author: assumed it will be the next release - but I'll have to update it after the release

> The package is available from version > 0.33.1.

Reviewer: Better to be concrete here. 
```suggestion
The package is available from version >= 0.34.0.
```
?

## Working with the Client API

How to instantiate a client:

```python
import lakefs_client
from lakefs_client import models
from lakefs_client.client import LakeFSClient

configuration = lakefs_client.Configuration()
configuration.username = 'AKIAIOSFODNN7EXAMPLE'
configuration.password = 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY'

> configuration.host = 'http://localhost:8000/api/v1'

Reviewer: Add a comment describing that these are the values for your lakeFS installation.
Might be trivial to us, but users often mix up these values.

client = Client(configuration)
``` 

## Using the generated client

Now that we have a client object, we can use it to interact with the API.

### Creating a repository

```python
repo = models.RepositoryCreation(name="example-repo", storage_namespace="s3://storage-bucket/repos/example-repo", default_branch="main")
client.repositories.create_repository(repo)
# output:
# {'creation_date': 1617532175,
#  'default_branch': 'main',
#  'id': 'example-repo',
#  'storage_namespace': 's3://storage-bucket/repos/example-repo'}
```

### Creating a branch, uploading files, committing changes

> List current branches:

Reviewer: Why `current`? Do you hold the state somewhere?
